File: src/lib/clipboard.ts
import { invoke } from "@tauri-apps/api/core";
import { capabilities } from "./stores/runtime.svelte.ts";

/** Read text from the system clipboard. Errors are intentionally preserved. */
export async function readText(): Promise<string> {
  return capabilities().nativeClipboard
    ? invoke<string>("clipboard_read")
    : navigator.clipboard.readText();
}

/** Write text to the system clipboard. Errors are intentionally preserved. */
export async function writeText(text: string): Promise<void> {
  if (capabilities().nativeClipboard) {
    await invoke<void>("clipboard_write", { text });
  } else {
    await navigator.clipboard.writeText(text);
  }
}

File: src-tauri/src/commands/clipboard.rs
use crate::error::AppResult;
use crate::platform::clipboard;

#[tauri::command]
pub async fn clipboard_read() -> AppResult<String> {
    clipboard::read_text().await
}

#[tauri::command]
pub async fn clipboard_write(text: String) -> AppResult<()> {
    clipboard::write_text(text).await
}
